dev/ci/scripts/utils/ci_utils.sh

```diff
@@ -177,6 +177,44 @@ function publish_logs() {
     echo "${URL}"
 }
 
+function publish_logs_from_file() {
+  # publish_logs_from_file
+  # Reads a file that lists relative file names (one per line) under a directory
+  # and publishes them using publish_logs.py. When more than one valid file is
+  # found, call the python utility with --multiple --format github for gist
+  # output formatting.
+  # Usage: publish_logs_from_file <ID> <list_file>
+  local PR_header="$1"
+  local list_file="$2"
+  local files=()
+
+  # Read the list file and build an array of existing full paths
+  while IFS= read -r line || [[ -n "${line}" ]]; do
+    # skip empty lines
+    [[ -z "${line// /}" ]] && continue
```

Comment:
Prefer this as an if block:

```suggestion
    if [[ -z "${line// /}" ]]; then
        continue
    fi
```